### PR TITLE
Gentoo support

### PR DESCRIPTION
**Pull request by gallart**
I added Gentoo support for puppet-spamassassin module + some code cleanup with variables instead of "strings". This will make other OS support easier.

### CODE REVIEW

**Comment by BillWeiss** [comment]
Could you clean up the failing tests?  Thanks for the PR!

**Comment by gallart** [comment]
Done ;)

File: manifests/init.pp [hunk 1]
```diff
@@ -112,10 +124,9 @@
     minute  => 10,
   }
 
-  service { 'spamassassin':
+  service { $sa_service:
     ensure  => $service_ensure,
     enable  => $service_enable,
     require => Package['spamassassin'],
-    pattern => 'spamd',
```
Comment (BillWeiss) [inline manifests/init.pp]: I don't love seeing this removed, but I don't see where it was actually used.  The debian init script (at least) provides a status command...

**Comment by BillWeiss** [comment]
Thanks!

**Pull request merged**